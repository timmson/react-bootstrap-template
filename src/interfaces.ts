export interface IHash {
	(name: string): number
}