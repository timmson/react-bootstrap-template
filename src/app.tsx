import React, {useState} from "react"
import {IHash} from "./interfaces"

type GreeterProps = {
    hash: IHash
}

const App = (props: GreeterProps) => {

	const [state, setState] = useState({name: ""})
	const update = (name) => setState({name: name})

	const greeting = state.name.length > 0 ?
		`Hello, ${state.name}! Your number is ${props.hash(state.name)}.` :
		"Please, start type your name"

	return (
		<div id="app" className="container-sm">
			<h2 className="text-center">Your fist React+Bootstrap app</h2>
			<div className="row mt-4">
				<div className="col-4 text-end align-content-center">
					<label htmlFor="username">Your name:</label>
				</div>
				<div className="col-8 align-content-center">
					<input className="form-control" placeholder="Enter your name ..." type="text" id="username"
						onKeyUp={(event) => update((event.target as HTMLInputElement).value)}/>
				</div>
			</div>
			<div className="row mt-4">
				<div className="col-4">
                    &nbsp;
				</div>
				<div className="col">
					<p className="my-style mt-2">{greeting}</p>
				</div>
			</div>
		</div>
	)
}

export default App
