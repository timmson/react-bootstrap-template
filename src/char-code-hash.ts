const CharCodeHash = (name: string) => {
	return Array.from(Array(name.length).keys())
		.map((i) => name.charCodeAt(i))
		.reduce((p, c) => p + c)
}

export default CharCodeHash
