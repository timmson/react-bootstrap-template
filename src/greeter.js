const React = require("react");
const PropTypes = require("prop-types");

class Greeter extends React.Component {

	static propTypes = {
		calc: PropTypes.object
	};

	constructor(props, context) {
		super(props, context);
		this.update = this.update.bind(this);
		this.state = {name: ""};
	}

	getGreeting() {
		return this.state.name.length > 0 ? `Hello, ${this.state.name}! Your number is ${this.props.calc.hash(this.state.name)}.` : "";
	}

	update(event) {
		this.setState({name: event.target.value});
	}

	render() {
		return (
			<div id="app" className="container">
				<h2 className="text-center">Your fist React+Bootstrap app</h2>
				<div className="row">
					<div className="col-4 text-right pt-4">
						<p>Your name:</p>
					</div>
					<div className="col align-middle">
						<input className="form-control mt-4" placeholder="Enter your name ..." type="text" id="username" onKeyUp={this.update}/>
					</div>
				</div>
				<div className="row mt-3">
					<div className="col-3">
						&nbsp;
					</div>
					<div className="col">
						<p className="my-style mt-2">{this.getGreeting()}</p>
					</div>
				</div>
			</div>
		);
	}
}

module.exports = Greeter;