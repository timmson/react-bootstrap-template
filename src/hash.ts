import {IHash} from "./interfaces"

const hash: IHash = (name: string) => {
	return Array.from(Array(name.length).keys())
		.map((i) => name.charCodeAt(i))
		.reduce((p, c) => p + c)
}

export default hash
