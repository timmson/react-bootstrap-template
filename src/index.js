require("./index.scss");
require("bootstrap");

const React = require("react");
const ReactDOM = require("react-dom");

const Greeter = require("./greeter");
const Calc = require("./calc");

ReactDOM.render(<Greeter calc={new Calc()}/>, document.getElementById("app"));