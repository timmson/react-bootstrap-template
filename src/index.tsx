import "./index.scss"
import "bootstrap"

import React from "react"
import ReactDOM from "react-dom"

import App from "./app"
import CharCodeHash from "./char-code-hash"

ReactDOM.render(<App hash={CharCodeHash}/>, document.getElementById("app"))