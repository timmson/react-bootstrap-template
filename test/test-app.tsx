import React from "react"
import * as renderer from "react-test-renderer"
import App from "../src/app"
import {IHash} from "../src/interfaces"

describe("Greeter should", () => {

	test("return greeting", () => {
		const mockHash: IHash = () => 0
		const component = renderer.create(<App hash={mockHash}/>)
		expect(component.toJSON()).toMatchSnapshot()
		component.unmount()
	})
})