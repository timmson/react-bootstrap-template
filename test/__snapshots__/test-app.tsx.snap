// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeter should return greeting 1`] = `
<div
  className="container-sm"
  id="app"
>
  <h2
    className="text-center"
  >
    Your fist React+Bootstrap app
  </h2>
  <div
    className="row mt-4"
  >
    <div
      className="col-4 text-end align-content-center"
    >
      <label
        htmlFor="username"
      >
        Your name:
      </label>
    </div>
    <div
      className="col-8 align-content-center"
    >
      <input
        className="form-control"
        id="username"
        onKeyUp={[Function]}
        placeholder="Enter your name ..."
        type="text"
      />
    </div>
  </div>
  <div
    className="row mt-4"
  >
    <div
      className="col-4"
    >
      Â 
    </div>
    <div
      className="col"
    >
      <p
        className="my-style mt-2"
      >
        Please, start type your name
      </p>
    </div>
  </div>
</div>
`;
